Locker/Locker/Locker.swift

```diff
@@ -0,0 +1,246 @@
+    static func shouldAddSecretToKeychainOnNextLogin(for uniqueIdentifier: String) -> Bool {
+        Locker.userDefaults?.bool(forKey: LockerHelpers.keyShouldAddSecretToKeychainOnNextLoginForUniqueIdentifier(uniqueIdentifier)) ?? false
+    }
+
+    @objc(setShouldAddSecretToKeychainOnNextLogin:forUniqueIdentifier:)
+    static func setShouldAddSecretToKeychainOnNextLogin(_ shouldAdd: Bool, for uniqueIdentifier: String) {
+        Locker.userDefaults?.set(shouldAdd, forKey: LockerHelpers.keyShouldAddSecretToKeychainOnNextLoginForUniqueIdentifier(uniqueIdentifier))
+    }
+}
+
+// MARK: - Data reset
+
+public extension Locker {
+
+    @objc(resetForUniqueIdentifier:)
+    static func reset(for uniqueIdentifier: String) {
+        Locker.userDefaults?.removeObject(forKey: LockerHelpers.keyDidAskToUseBiometricsIDForUniqueIdentifier(uniqueIdentifier))
+        Locker.userDefaults?.removeObject(forKey: LockerHelpers.keyShouldAddSecretToKeychainOnNextLoginForUniqueIdentifier(uniqueIdentifier))
+        Locker.userDefaults?.removeObject(forKey: LockerHelpers.keyBiometricsIDActivatedForUniqueIdentifier(uniqueIdentifier))
+        Locker.deleteSecret(for: uniqueIdentifier)
+    }
+}
+
+private extension Locker {
+    // added a function with a closure since we can't throw errors
+    // from async threads which is throwable
+    private static func setSecretForDevice(_ secret: String, for uniqueIdentifier: String, _ completion: @escaping ((LockerError?) throws -> Void)) {
+        let query: [CFString : Any] = [
+            kSecClass : kSecClassGenericPassword,
+            kSecAttrService : LockerHelpers.keyKeychainServiceName,
+            kSecAttrAccount : LockerHelpers.keyKeychainAccountNameForUniqueIdentifier(uniqueIdentifier)
+        ]
+
+        DispatchQueue.global(qos: .default).async {
+            // First delete the previous item if it exists
+            SecItemDelete(query as CFDictionary)
+
+            // Then store it
+            let errorRef: UnsafeMutablePointer<Unmanaged<CFError>?>? = nil
+            var flags: SecAccessControlCreateFlags
+            if #available(iOS 11.3, *) {
+                flags = .biometryCurrentSet
+            } else {
+                flags = .touchIDCurrentSet
+            }
+
+            let sacObject = SecAccessControlCreateWithFlags(
+                kCFAllocatorDefault,
+                kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly,
+                flags,
+                errorRef
+            )
+
+
+            guard let sacObject = sacObject, errorRef == nil, let secretData = secret.data(using: .utf8) else {
+                if let errorRef = errorRef {
+                    do {
+                        try completion(
+                            LockerError.accessControl("Unable to initialize access control: \(errorRef.pointee.debugDescription)")
+                        )
+                    } catch {
+                    }
```

Comment:
This is another example of why this block does not need to be throwable. Internally, we are not throwing anything so there is no need to define an interface as throwable.